### PR TITLE
Submit/3dslicer binary

### PR DESCRIPTION
**Pull request by MarcWeber**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Mainly adding this pull request so that others searching for it can find and run it. The SlicerApp-real runs and you can turn slices into STL which is all I cared about.

### CODE REVIEW

File: pkgs/servers/sql/mysql/5.6.x.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  passthru.mysqlVersion = "5.6";
+  passthru.mysqld_path = "bin";
+
+  passthru.mysql_initialize_datadir_cmd = {mysql, user, dataDir, ...}: ''
+    ${mysql}/bin/mysql_install_db "--user=${user} --datadir=${dataDir} --basedir=${mysql} ";
+  '';
+
+  meta = {
+    homepage = http://www.mysql.com/;
+    description = "The world's most popular open source database"; };
```
Comment (jonringer) [inline pkgs/servers/sql/mysql/5.6.x.nix:47]: please add license and maintainer

File: pkgs/servers/sql/mysql/5.6.x.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    url = https://cdn.mysql.com//Downloads/MySQL-5.6/mysql-5.6.48.tar.gz;
+    sha256 = "1hima34vnq92pdrhrm9acranplq0vjvahf3wg3siyk7ps6n27942";
+  };
+
+  preConfigure = stdenv.lib.optional stdenv.isDarwin ''
+    ln -s /bin/ps $TMPDIR/ps
+    export PATH=$PATH:$TMPDIR
+  '';
+
+  buildInputs = [ cmake bison ncurses openssl readline zlib perl ];
```
Comment (jonringer) [inline pkgs/servers/sql/mysql/5.6.x.nix:22]: tools that need to execute a build time should go in `nativeBuildInputs`
```suggestion
  nativeBuildInputs = [ cmake ];
  buildInputs = [ bison ncurses openssl readline zlib perl ];
```

File: pkgs/applications/misc/3dslicer-binary/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    ] 
+    ++ ( with pkgs.qt5; [qtbase qtsvg qtwebengine qtscript qtxmlpatterns]);
+
+  meta = with stdenv.lib; {
+    description = "";
+    license = licenses.bsd3; # like.
+    # https://github.com/Slicer/Slicer/blob/master/License.txt
+    # The 3D Slicer license below is a BSD style license, with extensions
+    # to cover contributions and other issues specific to 3D Slicer.
+    maintainers = [ ];
```
Comment (jonringer) [inline pkgs/applications/misc/3dslicer-binary/default.nix:52]: please add yourself

File: pkgs/applications/misc/3dslicer-binary/default.nix [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      pkgs.xorg.libXt
+      pkgs.libGLU
+
+      pkgs.mysql56
+      pkgs.postgresql.lib
+    ] 
+    ++ ( with pkgs.qt5; [qtbase qtsvg qtwebengine qtscript qtxmlpatterns]);
+
+  meta = with stdenv.lib; {
+    description = "";
```
Comment (jonringer) [inline pkgs/applications/misc/3dslicer-binary/default.nix:47]: please add description

File: pkgs/applications/misc/3dslicer-binary/default.nix [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    # to cover contributions and other issues specific to 3D Slicer.
+    maintainers = [ ];
+    platforms = [ "x86_64-linux" ];
+  };
+
+  installPhase = ''
+    mkdir -p $out
+    mv * $out
+    echo interpreter is $(cat $NIX_CC/nix-support/dynamic-linker)
+  '';
```
Comment (jonringer) [inline pkgs/applications/misc/3dslicer-binary/default.nix:60]: meta is usually last
```suggestion
  installPhase = ''
    mkdir -p $out
    mv * $out
    echo interpreter is $(cat $NIX_CC/nix-support/dynamic-linker)
  '';
  
  meta = with stdenv.lib; {
    description = "";
    license = licenses.bsd3; # like.
    # https://github.com/Slicer/Slicer/blob/master/License.txt
    # The 3D Slicer license below is a BSD style license, with extensions
    # to cover contributions and other issues specific to 3D Slicer.
    maintainers = [ ];
    platforms = [ "x86_64-linux" ];
  };
```

File: pkgs/applications/misc/3dslicer-binary/default.nix [hunk 6]
```diff
@@ -0,0 +1,61 @@
+{stdenv, pkgs}:
+/*
+The source distribution uses cmake and seems to depend on some downloads and
+git repositories thus might be quite some work to package and update.
+Thus using binaries for now.
+*/
+
+stdenv.mkDerivation rec {
+
+  name = "slicer-binary-4.10.2";
```
Comment (jonringer) [inline pkgs/applications/misc/3dslicer-binary/default.nix:10]: use pname
```suggestion
  pname = "slicer-binary";
  version = "4.10.2";
```

File: pkgs/applications/misc/3dslicer-binary/default.nix [hunk 7]
```diff
@@ -0,0 +1,61 @@
+{stdenv, pkgs}:
```
Comment (jonringer) [inline pkgs/applications/misc/3dslicer-binary/default.nix:1]: please avoid using pkgs, and instead import each dependency individually.

This allows for things like `.override()` to work correctly

File: pkgs/applications/misc/3dslicer-binary/default.nix [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  src = pkgs.fetchurl {
+    name = "Slicer-4.10.2-linux-amd64.tar.gz";
+    url = http://slicer.kitware.com/midas3/download?bitstream=1023242;
+    sha256 = "1x92120v055qjs6yapjjpcp9wp4ajd9qpgdzgqbnk1j2c8nl5c31";
+  };
+
+  libPath = stdenv.lib.makeLibraryPath (
+    [
+      pkgs.stdenv.cc.cc
+      pkgs.pkgs.xorg.libX11
```
Comment (jonringer) [inline pkgs/applications/misc/3dslicer-binary/default.nix:21]: please remove pkgs usage
```suggestion
      stdenv.cc.cc
      xorg.libX11
```

File: pkgs/applications/misc/3dslicer-binary/default.nix [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+      pkgs.zlib pkgs.bzip2 pkgs.sqlite
+    ] 
+    ++ ( with pkgs.qt5; [qtbase qtsvg qtwebengine qtscript qtxmlpatterns])
+  );
+
+
+  buildInputs = 
+    [
+      pkgs.gcc-unwrapped
+      pkgs.autoPatchelfHook
```
Comment (jonringer) [inline pkgs/applications/misc/3dslicer-binary/default.nix:32]: goes in native build inputs

File: pkgs/applications/misc/3dslicer-binary/default.nix [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+      pkgs.zlib pkgs.libxml2 pkgs.cups pkgs.pango pkgs.atk pkgs.gtk2 pkgs.glib pkgs.gdk-pixbuf
+      pkgs.zlib pkgs.bzip2 pkgs.sqlite
+    ] 
+    ++ ( with pkgs.qt5; [qtbase qtsvg qtwebengine qtscript qtxmlpatterns])
+  );
+
+
+  buildInputs = 
+    [
+      pkgs.gcc-unwrapped
```
Comment (jonringer) [inline pkgs/applications/misc/3dslicer-binary/default.nix:31]: available at build time anyway
```suggestion
```

File: pkgs/applications/misc/3dslicer-binary/default.nix [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+      pkgs.autoPatchelfHook
+
+      pkgs.stdenv.cc.cc
```
Comment (jonringer) [inline pkgs/applications/misc/3dslicer-binary/default.nix:34]:
```suggestion
```
also available

**Review by jonringer — COMMENTED** [review]
To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

in your case, the commit message should be:
```
mysql_5_6: init at 5.6.48
_3dslicer-binary: init at 4.10.2
```

**Comment by SuperSandro2000** [comment]
Closing because the author did not respond for quite a few months. Feel free to reopen.

**Pull request closed without merging**